src/data-management/store.ts

```diff
@@ -0,0 +1,61 @@
+import { create } from 'zustand';
+import { StorageService } from './storage-service';
+import { Note } from './types';
+
+export interface NotesState {
+    notes: Note[];
+    trash: Note[];
+
+    addNote: (newNote: Omit<Note, 'id' | 'createdAt' | 'modifiedAt'>) => void;
+    editNote: (
+        id: string,
+        updatedNote: Partial<Omit<Note, 'id' | 'createdAt' | 'modifiedAt'>>
+    ) => void;
+    deleteNote: (id: string, permanently?: boolean) => void;
+    getNotes: () => void;
+    getTrash: () => void;
+}
+
+const storageService = new StorageService();
+
+export const useNoteStore = create<NotesState>((set) => ({
+    notes: [],
+    trash: [],
+
+    getNotes: async () => {
+        const notes = await storageService.getNotes();
+        set({ notes });
+    },
+
+    getTrash: async () => {
+        const trash = await storageService.getNotes({ options: { includeDeleted: true } });
+        set({ trash });
+    },
+
+    addNote: async (newNote) => {
+        const addedNote = await storageService.addNote(newNote);
+        set((state) => ({ notes: [...state.notes, addedNote] }));
+    },
+
+    editNote: async (id, updatedNote) => {
```

Comment:
Would be great to have `pinNote` and `unpinNote` methods too. They will just accept note id.